code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingPushBuilder.java

```diff
@@ -210,30 +217,113 @@ private static void setSound(
     }
 
     /**
-     * Sets the large icon for the notification. If a large icon url is received from the payload,
-     * the image is downloaded and the notification style is set to BigPictureStyle. If large icon
-     * url is not received from the payload, default style is used for the notification.
+     * Sets the image for the notification then builds the notification. If a large icon url is
+     * received from the payload, the media is downloaded and the notification style is set to
+     * BigPictureStyle. If large icon url is not received from the payload, default style is used
+     * for the notification.
      *
      * @param notificationBuilder the notification builder
      * @param payload {@link MessagingPushPayload} the payload received from the push notification
+     * @return the built {@link Notification} object
      */
-    private static void setLargeIcon(
+    private static Notification buildNotification(
             final NotificationCompat.Builder notificationBuilder,
             final MessagingPushPayload payload) {
         // Quick bail out if there is no image url
-        if (StringUtils.isNullOrEmpty(payload.getImageUrl())) return;
-        Bitmap bitmap = MessagingPushUtils.download(payload.getImageUrl());
-
-        // Bail out if the download fails
-        if (bitmap == null) return;
-        notificationBuilder.setLargeIcon(bitmap);
-        NotificationCompat.BigPictureStyle bigPictureStyle =
-                new NotificationCompat.BigPictureStyle();
-        bigPictureStyle.bigPicture(bitmap);
-        bigPictureStyle.bigLargeIcon((Bitmap) null);
-        bigPictureStyle.setBigContentTitle(payload.getTitle());
-        bigPictureStyle.setSummaryText(payload.getBody());
-        notificationBuilder.setStyle(bigPictureStyle);
+        if (StringUtils.isNullOrEmpty(payload.getImageUrl())) return notificationBuilder.build();
+
+        // If the device is running on Android 34 or above and the image is a gif, download the gif
+        // and build the notification
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE
+                && MessagingPushUtils.isGifContent(payload.getImageUrl())) {
+            return downloadGifThenBuildNotification(notificationBuilder, payload);
+        } else {
+            final Bitmap bitmap = MessagingPushUtils.download(payload.getImageUrl());
+            // Bail out if the download fails
+            if (bitmap == null) return notificationBuilder.build();
+
+            final NotificationCompat.BigPictureStyle bigPictureStyle =
+                    new NotificationCompat.BigPictureStyle();
+            bigPictureStyle.bigPicture(bitmap);
```

Comment:
Do we also need to set setBigContentTitle and setSummaryText as before or is that not needed anymore?

Reply:
the title and content text set on the builder object will be used if big content title and summary text is not set.